### PR TITLE
Initial Post Restore compiler options processing 0.38

### PR DESCRIPTION
**Pull request by dsouzai**
Cherry-pick of https://github.com/eclipse-openj9/openj9/pull/16716

### CODE REVIEW

**Comment by dsouzai** [comment]
@mpirvu could you merge this once you merge https://github.com/eclipse-openj9/openj9/pull/16716

**Review by mpirvu — APPROVED** [review]

**Pull request merged**
